javascript/react/fundamentals/functional-components.md

```diff
@@ -35,9 +35,10 @@ aspects:
 
 As stated before, **React** components behave just like functions, taking `props` as input and returning **React elements**.
 
-Simple `component`s that don't have an internal `state` and don't make use of any **lifecycle** methods such as `constructor()` can be written as **functional components**.
+Simple components that don't have an internal `state` and don't make use of any **lifecycle** methods (such as `constructor()`) can be written as **functional components**.
```

Comment:
```suggestion
Components that don't make use of any **lifecycle** methods (such as `componentDidMount`) can be written just using functions and are usually called **functional components**.
```

Reply:
Hooks allow us to use state within functional components too.

In fact, this entire distinction is legacy now.

A more appropriate split is into something like:
- presentational component
- complex component

Where "presentational" components just render something and complex components perform some logic too, maybe render, maybe are used as HOC, etc.

A component in both groups can be written both as a class or a function.

Reply:
Once we create a standalone React topic, this insight will need to be renamed and mostly rewritten.